pkg/cmd/logs.go

```diff
@@ -0,0 +1,112 @@
+package cmd
+
+import (
+	"context"
+
+	"github.com/wercker/stern/kubernetes"
+
+	"github.com/cloud-native-application/rudrx/api/types"
+	cmdutil "github.com/cloud-native-application/rudrx/pkg/cmd/util"
+	"github.com/spf13/cobra"
+	sternCmd "github.com/wercker/stern/cmd"
+	"github.com/wercker/stern/stern"
+)
+
+func NewLogsCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = "logs <appname>"
+	cmd.Short = "Tail pods logs of an application"
+	cmd.Long = "Tail pods logs of an application"
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			ioStreams.Errorf("please specify the application name")
+		}
+		appName := args[0]
+		cmd.SetArgs([]string{appName + ".*"})
+		env, err := GetEnv(cmd)
+		if err != nil {
+			return err
+		}
+		namespace := env.Namespace
+		cmd.Flags().String("namespace", namespace, "")
+		return nil
+	}
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		config, err := sternCmd.ParseConfig(args)
+		if err != nil {
+			return err
+		}
+		ctx := context.Background()
+		if err := Run(ctx, config, ioStreams); err != nil {
+			return err
+		}
+		return nil
+	}
+	cmd.Annotations = map[string]string{
+		types.TagCommandType: types.TypeApp,
+	}
+	return cmd
+}
+
+// Run refer to the implementation at https://github.com/oam-dev/stern/blob/master/stern/main.go
+func Run(ctx context.Context, config *stern.Config, ioStreams cmdutil.IOStreams) error {
+	clientConfig := kubernetes.NewClientConfig(config.KubeConfig, config.ContextName)
+	clientSet, err := kubernetes.NewClientSet(clientConfig)
+	if err != nil {
+		return err
+	}
+	namespace := config.Namespace
+	added, removed, err := stern.Watch(ctx, clientSet.CoreV1().Pods(namespace), config.PodQuery, config.ContainerQuery, config.ExcludeContainerQuery, config.ContainerState, config.LabelSelector)
+	if err != nil {
+		return err
+	}
+	tails := make(map[string]*stern.Tail)
+	logC := make(chan string, 1024)
+
+	go func() {
+		for {
+			select {
+			case str := <-logC:
+				ioStreams.Infonln(str)
```

Comment:
The naming of function `Infonln()` is a little bit wired. How about directly using `fmt.Fprintf(os.Stdout, str)` instead?

Reply:
we'd better align with the whole project, you can try to refactor `cmdutil.IOStreams`